### PR TITLE
Add option for unread message badge

### PR DESCRIPTION
**Pull request by Gerben-T**
This PR adds a couple things to make it easier to recognize new unread messages.

When you have started a chat before and come back to the webpage at a later time, the library will now you have had a chat before and starts a "slow" polling. This polling is slower than the normal polling. 

Once there are new messages found the chat opens up by default (or when setting `window.parleySettings.interface.unreadMessagesAction = 0` explicitly) 

Instead of the chat opening you can also set `window.parleySettings.interface.unreadMessagesAction = 1` which will show an unread message badge with a counter in it.
See https://developers.parley.nu/docs/settings-2#unread-message-badge
The background color (red) and font color (white) are editable using the following css variables:
- `--parley-messenger-unread-messages-badge-background-color`
- `--parley-messenger-unread-messages-badge-font-color`

Further are some small improvements like:
- GET /messages now uses the /after:xxx filter to only fetch new messages instead of all messages
- Added more padding on the launcher button icon
- Scrolling to the latest message didn't work. Now it works as long as you didn't manually scroll yourself. There is a small margin at the bottom so you don't have to scroll all the way to the bottom to "re-enable" the automatic scrolling on new messages.
- Chat component has a key which re-renders it when the accountIdentification changes. Previously this wasn't an issue because we used GET (all) messages and changing the account identification cleared all your messages due to the endpoint returning `[]` for messages.
- Some more debug logging for important actions

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -68,6 +68,8 @@
 			--parley-close-alert-right-button-color: #070721;
 			--parley-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19);
 			--parley-image-loading-color: var(--parley-nav-background-color);
+			--parley-messenger-unread-messages-badge-background-color: red;
+			--parley-messenger-unread-messages-badge-font-color: white;
```
[the remark this answers is not in the document]
Comment (Gerben-T) [reply index.html]: done 026ea19e0275a70a70de8476b5c2b4cb8b81060c

File: src/UI/App.jsx [hunk 2]
```diff
@@ -460,6 +480,27 @@ export default class App extends React.Component {
...
 		 else
 			this.saveMessengerOpenState(this.state.messengerOpenState);
+
+		// Start slow polling if there was a chat started sometime before and chat starts minimized
+		if(messengerOpenState === MessengerOpenState.minimize) {
+			const devicePreviouslyRegistered = localStorage.getItem("deviceInformation") !== null;
+			if(devicePreviouslyRegistered !== null) {
+				Logger.debug("Starting slow polling service because device is previously registered");
+				if(this.PollingService.isRunning) {
+					Logger.debug("Polling service is running, stopping it because we only want slow polling at this moment");
```
[the remark this answers is not in the document]
Comment (Gerben-T) [reply src/UI/App.jsx:490]: done 9c36cac3887641c77f7c8187ec01752dd69634ab

[the remark this answers is not in the document]
Comment (Gerben-T) [reply src/UI/App.jsx]: done 9c36cac3887641c77f7c8187ec01752dd69634ab

File: src/UI/App.jsx [hunk 3]
```diff
@@ -705,45 +759,79 @@ export default class App extends React.Component {
...
+		}
+
 		this.PollingService.restartPolling();
 	};
 
 	handleNewMessage = (eventData) => {
 		// Keep track of all the message IDs, so we can show the
 		// chat when we received a new message
-		let foundNewMessages = false;
+		let _amountOfNewAgentMessagesFound = this.state.amountOfNewAgentMessagesFound;
```
[the remark this answers is not in the document]
Comment (Gerben-T) [reply src/UI/App.jsx:790]: dat is incorrect. deze if zorgt er voor dat we alleen agent messages tellen https://github.com/parley-messaging/web-library/blob/9c36cac3887641c77f7c8187ec01752dd69634ab/src/UI/App.jsx#L795-L795

File: src/UI/Conversation.jsx [hunk 4]
```diff
@@ -41,39 +45,72 @@ class Conversation extends Component {
...
+			// If there is no change we don't want to add them
+			// to prevent an unnecessary render.
+			const sortedMessages = Conversation.sortMessagesByID(messages);
+			if(JSON.stringify(sortedMessages) !== JSON.stringify(this.state.messages)) {
+				// Important that we make a clone of the state, otherwise we are updating the state directly
+				newState.messages = JSON.parse(JSON.stringify(this.state.messages));
+
+				// Add NEW messages to the newState
+				messages.forEach((message) => {
+					if(this.state.messages.filter(x => x.id === message.id).length > 0)
```
[the remark this answers is not in the document]
Comment (Gerben-T) [reply src/UI/Conversation.jsx:106]: done 9c36cac3887641c77f7c8187ec01752dd69634ab

File: src/UI/Conversation.jsx [hunk 5]
```diff
@@ -147,12 +184,37 @@ class Conversation extends Component {
...
+			// Client probably scrolled back down so we should reset the flag so we can scroll to new messages again
+			this.clientHasScrolledManually = false;
+		} else {
+			this.clientHasScrolledManually = true;
+		}
+	}
+
 	render() {
 		this.renderedDates = []; // Reset the rendered dates
+		const bodyRole = "feed"; // Used to keep jsx-a11y/no-static-element-interactions happy, not sure if this is the best fitting role...
```
[the remark this answers is not in the document]
Comment (Gerben-T) [reply src/UI/Conversation.jsx:205]: done 530884f5aedeead69902e1fc5a21bd75d2167a2f

File: src/UI/Launcher.jsx [hunk 6]
```diff
@@ -10,23 +10,42 @@ class Launcher extends Component {
 		const buttonType = "button";
 		const buttonId = "launcher";
 		const stateClass = this.props.messengerOpenState === MessengerOpenState.open ? styles["state-open"] : styles["state-minimize"];
-		const launcherClasses = `${styles.launcher} ${stateClass}`;
+		let launcherClasses = `${styles.launcher} ${stateClass}`;
 		const imgAltText = "icon";
 
+		// Add animation class when there are new unread messages
+		if(this.props.amountOfUnreadMessages > 0)
+			launcherClasses += ` ${styles.shake}`;
```
[the remark this answers is not in the document]
Comment (Gerben-T) [reply src/UI/Launcher.jsx:18]: Hij shaked altijd als er iets bij komt, dus
- niets => iets = shake
- iets => iets = shake

De prop `amountOfUnreadMessages ` krijgt namelijk een update en dan rerenderd react een deel van het component. Ik heb uiteindelijk een `key` toegevoegd zodat ie alles opnieuw rendered en dus de animatie weer doet https://github.com/parley-messaging/web-library/blob/08047ddca71d9e1956e2c0045df262556bc0462c/src/UI/Launcher.jsx#L29-L29

File: src/UI/App.jsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (daanleenders) [reply src/UI/App.jsx:790]: `...UnreadAgentMessages...` dan

File: src/UI/Launcher.jsx (hunk 6 above)
[the remark this answers is not in the document]
Comment (daanleenders) [reply src/UI/Launcher.jsx:18]: oooohw ja ik snap m. Door de key wordt het een nieuw dom element en die heeft dan die class en gaat shaken

**Review by daanleenders — APPROVED** [review]

**Pull request merged**